### PR TITLE
MARKET-88 Improve GitHub action templates

### PR DESCRIPTION
**Pull request by alexsuter**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,20 +16,13 @@ jobs:
...
       uses: actions/setup-java@v1.4.3
       with:
-        java-version: 11
-
-    - name: Build with Maven on master branch
-      if: github.ref == 'refs/heads/master'
-      run: mvn clean deploy --batch-mode
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        java-version: 8
```
Comment (ivy-rew) [inline .github/workflows/ci.yml:19]: 11

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -19,20 +16,13 @@ jobs:
...
+    - name: Build with Maven
       run: mvn clean verify --batch-mode
 
     - name: Publish Unit Test Results
       uses: EnricoMi/publish-unit-test-result-action@v1
       if: always()
       with:
-        files: "*/target/surefire-reports/*.xml"
\ No newline at end of file
+        files: "target/surefire-reports/*.xml"
```
Comment (ivy-rew) [inline .github/workflows/ci.yml:28]: was the beginning asteric necessary to match reports in sub-modules?

Comment (alexsuter) [reply .github/workflows/ci.yml]: you are right. thanks. eagle eye.

**Review by ivy-rew — APPROVED** [review]
Nice work @alexsuter . Cool build pipelines for free :1st_place_medal:

**Pull request merged**